feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/presentation/staking/start/setupStakingType/EditableStakingTypeItemFormatter.kt

```diff
@@ -0,0 +1,73 @@
+package io.novafoundation.nova.feature_staking_impl.presentation.staking.start.setupStakingType
+
+import io.novafoundation.nova.common.resources.ResourceManager
+import io.novafoundation.nova.feature_staking_impl.R
+import io.novafoundation.nova.feature_staking_impl.data.stakingType
+import io.novafoundation.nova.feature_staking_impl.domain.model.PayoutType
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.selection.RecommendableMultiStakingSelection
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.types.StakingTypeDetails
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupStakingType.model.ValidatedStakingTypeDetails
+import io.novafoundation.nova.feature_staking_impl.presentation.staking.start.common.MultiStakingTargetSelectionFormatter
+import io.novafoundation.nova.feature_staking_impl.presentation.staking.start.setupStakingType.adapter.EditableStakingTypeRVItem
+import io.novafoundation.nova.feature_wallet_api.domain.model.Asset
+import io.novafoundation.nova.feature_wallet_api.presentation.model.mapAmountToAmountModel
+import io.novafoundation.nova.runtime.ext.isDirectStaking
+import io.novafoundation.nova.runtime.ext.isPoolStaking
+
+class EditableStakingTypeItemFormatter(
+    private val resourceManager: ResourceManager,
+    private val multiStakingTargetSelectionFormatter: MultiStakingTargetSelectionFormatter
+) {
+
+    suspend fun format(
+        asset: Asset,
+        validatedStakingType: ValidatedStakingTypeDetails,
+        selection: RecommendableMultiStakingSelection
+    ): EditableStakingTypeRVItem? {
+        val stakingTarget = multiStakingTargetSelectionFormatter.formatForStakingType(selection)
+        val selectedStakingType = selection.selection.stakingOption.stakingType
+        val stakingType = validatedStakingType.stakingTypeDetails.stakingType
+
+        val title = when {
+            stakingType.isDirectStaking() -> resourceManager.getString(R.string.setup_staking_type_direct_staking)
+            stakingType.isPoolStaking() -> resourceManager.getString(R.string.setup_staking_type_pool_staking)
+            else -> return null
+        }
+
+        val imageRes = when {
+            stakingType.isDirectStaking() -> R.drawable.ic_pool_staking_banner_picture
+            stakingType.isPoolStaking() -> R.drawable.ic_direct_staking_banner_picture
+            else -> return null
+        }
```

Comment:
this can be combined:
``` kotlin

val title: String?
val imageRes: Int?

when {
 stakingType.isDirectStaking() -> {
 title = ...
 imageRes = ...
 }
stakingType.isPoolStaking() -> {
...
}
}